### PR TITLE
first pass at understanding the error messages in the cli, think that…

### PR DESCRIPTION
**Pull request by clee-cs**
… the catch blocks need to be moved from the `command` files to the `client` files or start returning promises in the client

Current error message that gets returned when trying to save an invalid dataset: ```Failed to save Dataset: 40000 Dataset must contain a valid connectionName```

### CODE REVIEW

File: src/commands/datasets.js [hunk 1]
```diff
@@ -89,8 +89,8 @@ module.exports.SaveDatasetsCommand = class SaveDatasetsCommand {
                printError(`Failed to save Dataset: ${response.status} ${response.message}`, options);
            }
        })
-       .catch((err) => {
-           printError(`Failed to save Dataset: ${err.status} ${err.message}`, options);
+       .catch((err) => { // don't think we ever get here..
+           printError(`Failed to save Datasets: ${err.status} ${err.message}`, options);
```
Comment (AgrimAsthana-CS) [inline src/commands/datasets.js]: Good catch!

**Comment by clee-cs** [comment]
Want to get yall's opinion on what to surface up before I wrap this in a function to standardize on it.. main question is which status code should we return? The http code or the internal code? Not sure if each service does it's own internal code or not..

**Comment by clee-cs** [comment]
Think I did this backwards compared to what Dave did.. just saw his commit to types and skills

File: src/commands/datasets.js (hunk 1 above)
Comment (clee-cs) [reply src/commands/datasets.js]: Since I added catch blocks in the client files, are all the catches in the commands no longer needed?

**Pull request merged**